DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/CourseSetup/AdminFieldsControllerTests.cs

```diff
@@ -194,6 +191,212 @@ public void AdminFieldBulkPost_updates_temp_data_and_redirects_to_edit()
+        {
+            var expectedPromptModel = new AddAdminFieldViewModel(1);
+            var initialTempData = new AddAdminFieldData(expectedPromptModel);
+            controller.TempData.Set(initialTempData);
+
+            // When
+            var result = controller.AddAdminField(1);
+
+            // Then
+            AssertAddAdminFieldViewModelIsExpectedModel(expectedPromptModel);
+            result.As<ViewResult>().Model.Should().BeOfType<AddAdminFieldViewModel>();
+        }
+
+        [Test]
+        public void AddAdminField_save_calls_service_and_redirects_to_index()
+        {
+            // Given
+            var model = new AddAdminFieldViewModel(100, 1, "Test");
+            const string action = "save";
+            var initialTempData = new AddAdminFieldData(model);
+            controller.TempData.Set(initialTempData);
+
+            A.CallTo(
+                () => courseAdminFieldsService.AddCustomPromptToCourse(
+                    100,
+                    101,
+                    0,
+                    1,
+                    "Test"
+                )
+            ).Returns(true);
+
+            // When
+            var result = controller.AddAdminField(1, model, action);
+
+            // Then
+            using (new AssertionScope())
+            {
+                A.CallTo(
+                    () => courseAdminFieldsService.AddCustomPromptToCourse(
+                        100,
+                        101,
+                        0,
+                        1,
+                        "Test"
+                    )
+                ).MustHaveHappened();
+                controller.TempData.Peek<AddAdminFieldData>().Should().BeNull();
+                result.Should().BeRedirectToActionResult().WithActionName("AdminFields");
+            }
+        }
+
+        [Test]
+        public void AddAdminField_calls_service_and_redirects_to_error_on_failure()
+        {
+            // Given
+            var model = new AddAdminFieldViewModel(100, 1, "Test");
+            const string action = "save";
+            var initialTempData = new AddAdminFieldData(model);
+            controller.TempData.Set(initialTempData);
+
+            A.CallTo(
+                () => courseAdminFieldsService.AddCustomPromptToCourse(
+                    100,
+                    101,
+                    0,
+                    1,
+                    "Test"
+                )
+            ).Returns(false);
+
+            // When
+            var result = controller.AddAdminField(100, model, action);
+
+            // Then
+            using (new AssertionScope())
+            {
+                A.CallTo(
+                    () => courseAdminFieldsService.AddCustomPromptToCourse(
+                        100,
+                        101,
+                        0,
+                        1,
+                        "Test"
+                    )
+                ).MustHaveHappened();

```

Comment:
We're already mocking this to return true/false in these two tests, which makes its inclusion in the assertion unnecessary (and also the tests will get more difficult to read/scan over). If you wanted you could make a separate test for checking that the mock gets called once with the correct values, but I think it would probably classify as "overtesting" the controller :)